pkg/get/download.go

```diff
@@ -31,6 +32,24 @@ func Download(tool *Tool, arch, operatingSystem, version string, downloadMode in
 		return "", "", err
 	}
 
+	// Check user provided version is valid on dry run
+	if len(version) > 1 && dry_run {
+		res, err := http.Head(downloadURL)
```

Comment:
Can you extract this code block into its own method with a descriptive name?

I'd also like to see you use http.NewRequest instead of the shortcut http.Head. See how we make HTTP requests elsewhere in the code for an example.

You'll then want `http. MethodHead`, https://pkg.go.dev/net/http#pkg-constants